fn/func.py

```diff
@@ -80,5 +81,8 @@ def _curried(*args, **kwargs):
         if count == len(spec.args) - len(args):
             return func(*args, **kwargs)
 
-        return curried(partial(func, *args, **kwargs))
+        para_func = partial(func, *args, **kwargs)
+        update_wrapper(para_func, f)
```

Comment:
`f` or `func`? 

`f` was the iterator variable in the function loop, above. I suspect it's not the right choice here.